benchmarks/nemo/nemolite2d/manual_versions/psykal_sycl/time_step_mod.f90

```diff
@@ -23,14 +23,15 @@ subroutine wrapper_c_invoke_time_step( &
             istp, internal_xstart, internal_xstop, internal_ystart, &
             internal_ystop, width, total_size, rdt, cbfr, visc, omega, d2r, g &
         ) bind (C, name="c_invoke_time_step")
-            use iso_c_binding
+            use iso_c_binding, only: c_double, c_int, c_ptr, c_intptr_t
             real(kind=c_double), intent(inout), dimension(*) :: ssha_t, &
                 sshn_t, sshn_u, sshn_v, hu, hv, un, vn, ua, ht, ssha_u, va, &
                 ssha_v, area_t, area_u, area_v, dx_u, dx_v, dx_t, dy_u, dy_v, &
                 dy_t, gphiu, gphiv
-            integer(c_intptr_t), intent(inout) :: ssha_t_dp, sshn_t_dp, &
+            type(c_ptr), intent(inout) :: ssha_t_dp, sshn_t_dp, &
                 sshn_u_dp, sshn_v_dp, hu_dp, hv_dp, un_dp, vn_dp, ua_dp, &
-                ht_dp, ssha_u_dp, va_dp, ssha_v_dp, tmask_dp, area_t_dp, &
+                ht_dp, ssha_u_dp, va_dp, ssha_v_dp
+            integer(c_intptr_t), intent(inout) :: tmask_dp, area_t_dp, &
```

Comment:
Just to check, why are some of these now c_intptr_t while the fields are c_ptr?

Reply:
Updated all to c_ptr now